cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunner.java

```diff
@@ -101,8 +101,34 @@ public PullAppResponse<?> pull(ApplicationReference appRef, RepositoryConfig rep
   }
 
   @Override
-  public RepositoryAppsResponse list(NamespaceId namespace, RepositoryConfig repoConfig)
+  public RepositoryAppsResponse list(NameSpaceRepository nameSpaceRepository)
     throws AuthenticationConfigException, NotFoundException {
-    // TODO: CDAP-20357, list applications in task worker
-    throw new UnsupportedOperationException("Not implemented");  }
+    try {
+      RunnableTaskRequest request =
+        RunnableTaskRequest.getBuilder(ListAppsTask.class.getName()).withParam(GSON.toJson(nameSpaceRepository)).build();
+
+      LOG.trace("Listing applications for namespace {} in linked repository", nameSpaceRepository.getNamespaceId());
+      byte[] result = remoteTaskExecutor.runTask(request);
+      return GSON.fromJson(new String(result, StandardCharsets.UTF_8), RepositoryAppsResponse.class);
+    } catch (RemoteExecutionException e) {
+      // Getting the actual RemoteTaskException
+      // which has the root cause stackTrace and error message
+      RemoteTaskException remoteTaskException = e.getCause();
```

Comment:
Seems like duplicated code from other remote task execution methods. Please refactor.

Reply:
I think @de-lan opened a jira to improve this across the entire code base for whatever uses the RemoteTaskExecutor, but for the time being we can have just a common method in this class for handling the exception. For example, a method where you pass in the RemoteExecutionException and the expected exception classes.

Reply:
yes, we have a ticket for this: https://cdap.atlassian.net/browse/CDAP-20418